### DISCUSSION TITLE
Unregister not working with Controller

### DISCUSSION THREAD

**Discussion by StevenVerbiest** [react-hook-form/react-hook-form#5385] [Q&A]
I'm using react hook form `v6.15.4` and I'n unable to unregister a Material UI TextField implemented in a `Controller` wrapper.
Unregistering a "normal" input works just fine.

I've added a codesandbox to illustrate te problem:
https://codesandbox.io/s/react-hook-form-unregister-with-controller-5c03l?file=/src/App.tsx

**Expected behavior**
Using `unregister(fieldname)` when this field is implemented in a `<Controller>` should omit this fields data in the submit handler.

**Comment by bluebill1049** [comment]
This is due to Controller is getting re-register again during render, I would unmount that Controller as well.

**Reply by StevenVerbiest** [reply]
What do you mean exactly by "unmount that Controller"? How would you do that? At what point should it be unmounted?

**Reply by bluebill1049** [reply]
```
const [show, setShow] = React.useState(false)

const onClick = () => {
  unregister('test');
  setShow(false);
}

{show && <Controller />}
```

also depend on the mode that you are using, if shouldUnregister: true, then when the Controller gets unmounted, it will just unregister that form state for you.

**Reply by StevenVerbiest** [reply]
I'm afraid the input should be visible at all times. So this is not an option I think. 
The reason I looked into unregistering the component is because I wanted to take advantage of the the built-in validation, but remove the data from submit. 

But after some extra investigation I think this approach is not possible, even with "regular" input, because the field value will be removed, but the validation will still be triggered, with an empty value.

I guess in my case the options left are:
* Not using the input within react hook form and trigger the validation manually
* Removing the fields data in the submit handler

Not sure if there's another option I overlooked? 

Thanks for the replies bye te way!

**Reply by bluebill1049** [reply]
let me understand what you try to achieve better, you want to remove the input value or just skip the validation?

**Reply by StevenVerbiest** [reply]
I want to remove the input value.

**Reply by bluebill1049** [reply]
if you unmount that Controller with a piece of state, that value should be removed as well right?
